packages/runtime-core/src/renderer.ts

```diff
@@ -1217,7 +1217,7 @@ function baseCreateRenderer(
     }
 
     // resolve props and slots for setup context
-    if (!(__COMPAT__ && compatMountInstance)) {
+    if (!compatMountInstance) {
```

Comment:
```
const compatMountInstance =
      __COMPAT__ && initialVNode.isCompatRoot && initialVNode.component
```
This does not seem to be simplifyable

Reply:
maybe the __COMPAT__ check in if (!(__COMPAT__ && compatMountInstance)) is redundant ?

Reply:
Maybe the __COMPAT__ check in if (!(__COMPAT__ && compatMountInstance)) is redundant?

Reply:
@stayhpjinng No is not redundant, since `__COMPAT__` will be replaced and statically check at build time, if `false` the if branch will be removed. 

If you remove `__COMPAT__` the if branch will always be present regardless of the `compatMountInstance` being false.